File: test-suite/golden-tests/arrow-immutable-pr/src.py
import pyarrow as pa

def makeLargeTable(n):
    return pa.RecordBatch.from_pydict({
        "idx": list(range(n)),
        "value": [float(i) * 0.5 for i in range(n)]
    })

def makeIndices(n):
    return list(range(n))

def sumReals(xs):
    return sum(xs)

File: test-suite/golden-tests/arrow-immutable-rp/src.R
makeLargeTable <- function(n) {
  arrow::record_batch(
    idx = 0L:(n - 1L),
    value = as.double(0:(n - 1)) * 0.5
  )
}

makeIndices <- function(n) {
  as.integer(0:(n - 1))
}

sumReals <- function(xs) {
  sum(xs)
}
